torch/CMakeLists.txt

```diff
@@ -476,15 +476,237 @@ install(FILES
+    ${TORCH_SRC_DIR}/csrc/utils/tensor_layouts.cpp
+    ${TORCH_SRC_DIR}/csrc/utils/tensor_list.cpp
+    ${TORCH_SRC_DIR}/csrc/utils/tensor_new.cpp
+    ${TORCH_SRC_DIR}/csrc/utils/tensor_numpy.cpp
+    ${TORCH_SRC_DIR}/csrc/utils/tensor_types.cpp
+    ${TORCH_SRC_DIR}/csrc/utils/tuple_parser.cpp
+    )
+
+  set(TORCH_PYTHON_INCLUDE_DIRECTORIES
+    ${PYTHON_INCLUDE_DIR}
+
+    ${TORCH_ROOT}
+    ${TORCH_ROOT}/aten/src
+    ${TORCH_ROOT}/aten/src/TH
+
+    ${CMAKE_BINARY_DIR}
+    ${CMAKE_BINARY_DIR}/aten/src
+    ${CMAKE_BINARY_DIR}/caffe2/aten/src
+    ${CMAKE_BINARY_DIR}/caffe2/aten/src/TH
+    ${CMAKE_BINARY_DIR}/caffe2/aten/src/THC
+    ${CMAKE_BINARY_DIR}/third_party
+    ${CMAKE_BINARY_DIR}/third_party/onnx
+
+    ${TORCH_ROOT}/third_party/build/nccl/include
+    ${TORCH_ROOT}/third_party/gloo
+    ${TORCH_ROOT}/third_party/onnx
+    ${TORCH_ROOT}/third_party/protobuf/src
+    ${TORCH_ROOT}/third_party/pybind11/include
+
+    ${TORCH_SRC_DIR}/csrc
+    ${TORCH_SRC_DIR}/lib
+    )
+
+  if (MSVC)
+    list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES
+      ${TORCH_SRC_DIR}/lib/libshm_windows)
+  else()
+    list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES
+      ${TORCH_SRC_DIR}/lib/libshm)
+  endif()
+
+  set(TORCH_PYTHON_LINK_LIBRARIES
+    torch
+    shm)
+
+  set(TORCH_PYTHON_COMPILE_DEFINITIONS)
+
+  set(TORCH_PYTHON_COMPILE_OPTIONS)
+
+  set(TORCH_PYTHON_LINK_FLAGS)
+
+  if (MSVC)
+    list(APPEND TORCH_PYTHON_LINK_FLAGS " /NODEFAULTLIB:LIBCMT.LIB")
+    list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${PYTHON_LIBRARIES})
+  elseif (APPLE)
+    list(APPEND TORCH_PYTHON_LINK_FLAGS " -undefined dynamic_lookup")
+  else()
+    list(APPEND TORCH_PYTHON_COMPILE_OPTIONS
+      -fno-strict-aliasing
+      -Wno-write-strings
+      -Wno-strict-aliasing)
+  endif()
+
+  if (USE_CUDA)
+    list(APPEND TORCH_PYTHON_SRCS
+      ${TORCH_SRC_DIR}/csrc/cuda/Module.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/Storage.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/Stream.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/utils.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/python_comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/serialization.cpp
+      ${TORCH_SRC_DIR}/csrc/nn/THCUNN.cpp
+      )
+    list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_CUDA)
+
+    if(MSVC)
+      list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${NVTOOLEXT_HOME}/lib/x64/nvToolsExt64_1.lib)
+      list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES "${NVTOOLEXT_HOME}/include")
+      list(APPEND TORCH_PYTHON_LINK_FLAGS " /FORCE:UNRESOLVED")
+    elseif(APPLE)
+      list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDA_TOOLKIT_ROOT_DIR}/lib/libnvToolsExt.dylib)
+    else()
+      list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDA_TOOLKIT_ROOT_DIR}/lib64/libnvToolsExt.so)
+    endif()
+  endif()
+
+  if (USE_CUDNN)
+    list(APPEND TORCH_PYTHON_COMPILE_DEFINITIONS USE_CUDNN)
+    list(APPEND TORCH_PYTHON_INCLUDE_DIRECTORIES ${CUDNN_INCLUDE_DIR})
+    list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDNN_LIBRARY})
```

Comment:
you need to prepend this, not append (see my comment in setup.py about double-checking on this)

Reply:
done